coordinator/src/main/scala/filodb.coordinator/RowSource.scala

```diff
@@ -85,74 +75,86 @@ trait RowSource extends Actor with StrictLogging {
 
   def reading: Receive = {
     case GetMoreRows =>
```

Comment:
A lot of code to reside in a receive.. personally I'd make it clean, save it to a function and place the processing in that function.